ompi/mca/coll/ucx/coll_ucx_module.c

```diff
@@ -0,0 +1,157 @@
+    for (rank_idx = 0; rank_idx < args.member_count; rank_idx++) {
+    	struct ompi_proc_t *rank_iter =
+    			(struct ompi_proc_t*)ompi_comm_peer_lookup(comm, rank_idx);
+        rank_iter->proc_endpoints[OMPI_PROC_ENDPOINT_TAG_COLL] = NULL;
+        if (rank_idx == my_idx) {
+            args.distance[rank_idx] = UCG_GROUP_MEMBER_DISTANCE_SELF;
+        } else if (OPAL_PROC_ON_LOCAL_SOCKET(rank_iter->super.proc_flags)) {
+            args.distance[rank_idx] = UCG_GROUP_MEMBER_DISTANCE_SOCKET;
+        } else if (OPAL_PROC_ON_LOCAL_HOST(rank_iter->super.proc_flags)) {
+            args.distance[rank_idx] = UCG_GROUP_MEMBER_DISTANCE_HOST;
+        } else {
+            args.distance[rank_idx] = UCG_GROUP_MEMBER_DISTANCE_NET;
+        }
+    }
+
+    /* TODO: use additional info, such as OMPI_COMM_IS_CART(comm) */
+    /* TODO: add support for comm->c_remote_comm  */
+    /* TODO: add support for sparse group storage */
+    error = ucg_group_create(mca_coll_ucx_component.ucg_worker, &args, &module->ucg_group);
+
+    /* Examine comm_new return value */
+    if (error != UCS_OK) {
+        COLL_UCX_ERROR("ucg_new failed: %s", ucs_status_string(error));
+        return OMPI_ERROR;
+    }
+
+    ucs_list_add_tail(&mca_coll_ucx_component.group_head, &module->ucs_list);
+    return OMPI_SUCCESS;
+}
+
+/*
+ * Initialize module on the communicator
+ */
+static int mca_coll_ucx_module_enable(mca_coll_base_module_t *module,
+                                      struct ompi_communicator_t *comm)
+{
+    mca_coll_ucx_module_t *ucx_module = (mca_coll_ucx_module_t*) module;
+    int rc;
+
+    /* prepare the placeholder for the array of request* */
+    module->base_data = OBJ_NEW(mca_coll_base_comm_t);
+    if (NULL == module->base_data) {
+        return OMPI_ERROR;
+    }
+
+    rc = mca_coll_ucg_create(ucx_module, comm);
+    if (rc != OMPI_SUCCESS)
+        return rc;
+
+    COLL_UCX_FREELIST_INIT(&mca_coll_ucx_component.persistent_ops,
+                          mca_coll_ucx_persistent_op_t,
+                          128, -1, 128);
+
+    COLL_UCX_VERBOSE(1, "UCX Collectives Module initialized");
+    return OMPI_SUCCESS;
+}
+
+static int mca_coll_ucx_ft_event(int state) {
+    if(OPAL_CRS_CHECKPOINT == state) {
+        ;
+    }
+    else if(OPAL_CRS_CONTINUE == state) {
+        ;
+    }
+    else if(OPAL_CRS_RESTART == state) {
+        ;
+    }
+    else if(OPAL_CRS_TERM == state ) {
+        ;
+    }
+    else {
+        ;
+    }
+
+    return OMPI_SUCCESS;
+}
+
+static void mca_coll_ucx_module_construct(mca_coll_ucx_module_t *module)
+{
+    size_t nonzero = sizeof(module->super.super);
+    memset((void*)module + nonzero, 0, sizeof(*module) - nonzero);
+
+    module->super.coll_module_enable  = mca_coll_ucx_module_enable;
+    module->super.ft_event            = mca_coll_ucx_ft_event;
+    module->super.coll_allreduce      = mca_coll_ucx_allreduce;
+//  module->super.coll_iallreduce     = mca_coll_ucx_iallreduce;
```

Comment:
Please don't commit commented-out code.  If these are function pointers for code that isn't ready yet, it probably shouldn't be committed.